### PR TITLE
Perf/txn-data-mobx (#1265)

### PR DESCRIPTION
**Pull request by matthew-garrett**
* build: updates geoip block

build: removes notional context

build: removes logs

* build: adds hasNoteOrSNote check back
build: removes notional context

build: moves useTransaction logic to WalletStore

### CODE REVIEW

File: packages/features/portfolio/src/portfolio-feature-shell.tsx [hunk 1]
```diff
@@ -45,12 +45,10 @@ export const PortfolioFeatureShell = observer(() => {
   const network = useSelectedNetwork();
   const isAccountLoading = useAccountLoading();
   const isAcctAndBalanceReady = useAccountAndBalanceReady(network);
-  const { hasNoteOrSNote } = usePortfolioNOTETable();
```
Comment (jeffywu) [inline packages/features/portfolio/src/portfolio-feature-shell.tsx]: Were we supposed to keep this in or no?

File: packages/notionable/src/stores/wallet-store.ts [hunk 2]
```diff
@@ -62,8 +71,12 @@ export const WalletModel = types
     networkAccounts: types.optional(types.map(AccountPortfolioModel), {}),
     totalPoints: types.maybe(types.number),
     sentTransactions: types.optional(types.array(SentTransactionModel), []),
-    completedTransactions: types.optional(CompletedTransactionModel, {}),
+    completedTransactions: types.maybe(TransactionReceiptModel),
```
Comment (jeffywu) [inline packages/notionable/src/stores/wallet-store.ts]: Save this as an array of completedTransactions

File: packages/notionable/src/stores/wallet-store.ts [hunk 3]
```diff
@@ -133,6 +152,94 @@ export const WalletModel = types
...
+        } else if (receipt.status === 0) {
+          setTransactionStatus(TransactionStatus.REVERT);
+          trackEvent(TRACKING_EVENTS.TXN_ERROR, {
+            url: window.location.pathname,
+            transactionLabel,
+            selectedNetwork: self.userWallet?.selectedChain,
+          });
+        }
+        setCompletedTransactions(receipt);
+      } catch (error: any) {
```
Comment (jeffywu) [inline packages/notionable/src/stores/wallet-store.ts]: Just remove the `error: any` here, it is unused

File: packages/features/portfolio/src/portfolio-feature-shell.tsx (hunk 1 above)
Comment (matthew-garrett) [reply packages/features/portfolio/src/portfolio-feature-shell.tsx]: We are keeping this

**Pull request merged**